routers/web/repo/view.go

```diff
@@ -518,7 +518,7 @@ func renderFile(ctx *context.Context, entry *git.TreeEntry, treeLink, rawLink st
 				Metas:     ctx.Repo.Repository.ComposeDocumentMetas(),
 				GitRepo:   ctx.Repo.GitRepo,
 			}, rd, &result)
-			if err != nil {
+			if err != nil && err != io.EOF {
```

Comment:
This may be wrong for other renderers. You could check in the csv renderer for `io.EOF` and treat it as no error.

Reply:
Yeah, I was thinking of looking at that a bit deeper, but if all it reads for any renderer is an EOF marker should we really through a big 500 error rather than show an empty file that they can deal with?

Reply:
I don't think a renderer should be returning io.EOF at all.

Reply:
Ok, will just make sure the CSV renderer isn't doing that. I do think we should make sure all Renders don't do that, but will let that be a separate issue

Reply:
Ok, I have changed to just making sure the two places that the CSV Render reads the file to just return nil if it encounters only an EOF marker.